.github/GSOC.md

```diff
@@ -0,0 +1,123 @@
+
+## :technologist: Your Proposal
+
+Your proposal is your opportunity to show us that you are the right person for the job. To this end,
+we have provided a list of things that we would like to see in your proposal. We are not looking for
+a long proposal, but we are looking for a well-thought-out proposal that shows us that you have
+taken the time to understand the project and have a clear plan for how you will contribute.
+
+### Proposal Outline
+
+#### Introduction
+
+All projects aim to solve a problem or improve something. Start your proposal by introducing the
+problem you are trying to solve or the improvement you are trying to make. This is your opportunity
+to show us that you understand the problem, why it is important, and why it is worth solving.
+Include any relevant background information that will help us understand the problem, like
+statistics, examples, references to existing issues, etc.
+
+#### Project Goals
+
+This should be concise and to the point. What are you trying to accomplish? What are the
+deliverables? What are the success criteria? What are the key milestones? Effective goals are
+specific, measurable, and achievable. This section may be a bulleted list or outline.
+
+#### Implementation Plan
+
+Describe your solution to the problem. This should include a high-level overview of the
+architecture, a list of tasks, and a list of potential challenges, as well as some technical
+details. We want to see that you have thought through the problem and can visualize how you will
+solve it.
+
+#### Timeline
+
+Show us that you can break down your project into manageable tasks and that you have a clear plan
+for how you will accomplish your goals. This should be a timeline that shows what you will be
+working on each week. It should include milestones and deliverables. An effective way to present
+this is in a table or a Gantt chart.
+
+#### About You
+
+Tell us about yourself. What is your background? What are your interests? What are your hobbies?
+What are your career goals? What do you hope to get out of GSoC? What do you hope to contribute to
+Checkstyle? This is your opportunity to show us that you are a good fit for the project and that
+you are excited to work with us. **You must include a list of links to your Checkstyle contributions
+to be considered for GSoC; this is one of the most important selection criteria.**
+
+### Proposal Tips
+
+- **Be Specific.** We want to see that you have thought through the problem and
+  how you will solve it.
+- **Be Realistic.** We are looking for a proposal that is achievable in the time frame of GSoC.
+  We want to see that you have a clear plan.
+- **Be Professional.** We are looking for a proposal that is well-written and professional.
+- **One Proposal <-> One Project.** Do not create a proposal that spans multiple projects.
+- **Proposal Title** If creating a proposal based on our project ideas, please use the title of the
```

Comment:
Why is this one a tip and not a requirement? There is nothing here that seems subjective. I consider a tip something that is beneficial but can be ignored (but probably unwise).

Reply:
There should be no IF. All matching to existing project titles.

If someone is so experienced in our project to propose new idea, this person most likely experienced enough to understand that such ideas better to bring in advance, and there is always room to change list of projects, but extra efforts will be required to persuade other to take it.